pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,136 @@
+	// be empty or match the first entry in the list.
+	// Default is 10.0.0.0/16 for all platforms other than libvirt.
+	// For libvirt, the default is 192.168.126.0/24.
+	//
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+
+	// ClusterNetwork is the list of IP address pools for pods.
+	// Default is 10.128.0.0/14 and a host prefix of /23.
+	//
+	// +optional
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork,omitempty"`
+
+	// ServiceNetwork is the list of IP address pools for services.
+	// Default is 172.30.0.0/16.
+	// NOTE: currently only one entry is supported.
+	//
+	// +kubebuilder:validation:MaxItems=1
+	// +optional
+	ServiceNetwork []IPNet `json:"serviceNetwork,omitempty"`
+}
+
+// MachineNetworkEntry is a single IP address block for node IP blocks.
+type MachineNetworkEntry struct {
+	// CIDR is the IP block address pool for machines within the cluster.
+	CIDR IPNet `json:"cidr"`
+}
+
+// ClusterNetworkEntry is a single IP address block for pod IP blocks. IP blocks
+// are allocated with size 2^HostSubnetLength.
+type ClusterNetworkEntry struct {
+	// CIDR is the IP block address pool.
+	CIDR IPNet `json:"cidr"`
+
+	// HostPrefix is the prefix size to allocate to each node from the CIDR.
+	// For example, 24 would allocate 2^8=256 adresses to each node. If this
+	// field is not used by the plugin, it can be left unset.
+	// +optional
+	HostPrefix int32 `json:"hostPrefix,omitempty"`
+}
+
+// Proxy defines the proxy settings for the cluster.
+// At least one of HTTPProxy or HTTPSProxy is required.
+type Proxy struct {
+	// HTTPProxy is the URL of the proxy for HTTP requests.
+	// +optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
+
+	// HTTPSProxy is the URL of the proxy for HTTPS requests.
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+
+	// NoProxy is a comma-separated list of domains and CIDRs for which the proxy should not be used.
+	// +optional
+	NoProxy string `json:"noProxy,omitempty"`
+}
+
+type ClusterProgressInfo struct {
+	// progress info
+	ProgressInfo string `json:"progressInfo,omitempty"`
+	// Time at which the cluster install progress was last updated.
+	LastProgressUpdateTime *metav1.Time `json:"lastProgressUpdateTime,omitempty"`
+}
+
+type HostNetwork struct {
+	// cidr
+	Cidr string `json:"cidr,omitempty"`
+	// host ids
+	HostIds []string `json:"hostIDs"`
+}
+
+// ClusterStatus defines the observed state of Cluster
+type ClusterStatus struct {
+	State                        string              `json:"state,omitempty"`
+	StateInfo                    string              `json:"stateInfo,omitempty"`
+	HostNetworks                 []HostNetwork       `json:"hostNetworks,omitempty"`
+	InstallationStartTime        string              `json:"installationStartTime,omitempty"`
+	InstallationCompletionTime   *metav1.Time        `json:"installationCompletionTime,omitempty"`
+	Hosts                        int                 `json:"hosts,omitempty"`
```

Comment:
this should probably be a list of objects or list of references to another object that have info on these hosts.

and instead of just a count, we should keep count of useful categories of host, like total, ready, validating, failed etc..

Reply:
@mhrivnak Suggested this, the user can list hosts by cluster label

Reply:
Suspect we should drop this one, finding the agents/hosts for a cluster should be handled by some other association. We should avoid per host entries within a CRD (size concerns, large updates/etcd writes) if possible.

Reply:
Generally there's value in understanding how many agents are expected vs. present. For example, a replica set will show you the number of observed pods in its status, and additionally how many are ready vs. available.

That said, this number becomes less relevant after installation is started, and especially after it is complete. It may be more useful to report `agentsExpected`, `agentsDiscovered`, and `agentsReady` or something like that.

Reply:
Makes sense to me

Reply:
I have replaced hosts with the above three distinctions.